File: 백준/Silver/1940. 주몽/주몽.java
import java.util.*;

public class Main {
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int m = sc.nextInt();
        int arr[] = new int [n];

        for(int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }
        Arrays.sort(arr);

        int cnt = 0;
        int start = 0;
        int end = n - 1;

        while(start < end){
            if(arr[start] + arr[end] < m)
                start++;
            else if(arr[start] + arr[end] > m)
                end--;
            else{
                cnt++;
                start++;
                end--;
            }
        }
        System.out.println(cnt);
	}

}

File: 백준/Silver/1940. 주몽/주몽.py
import sys

n = int(sys.stdin.readline())
m = int(sys.stdin.readline())
num = list(map(int, sys.stdin.readline().split()))

num.sort()

cnt = 0
start = 0
end = len(num) - 1

while start < end:
    if num[start] + num[end] == m:
        start += 1
        end -= 1
        cnt += 1
    elif num[start] + num[end] < m:
        start += 1
    elif num[start] + num[end] > m:
        end -= 1
print(cnt)